tests/ui/components/navigation/navigationcomponent.js

```diff
@@ -102,3 +112,70 @@ describe('navigation component configuration', () => {
     expect(tab).toMatchObject(expected);
   });
 });
+
+describe('navigation tab links are correct', () => {
+  let COMPONENT_MANAGER;
+
+  const defaultConfig = {
+    container: '#test-component',
+    verticalKey: 'verticalKey',
+    verticalPages: [
+      { verticalKey: 'vertical', url: './vertical', label: 'Vertical' },
+      { verticalKey: 'facets', url: './facets', label: 'Facets' }
+    ]
+  };
+
+  beforeEach(() => {
+    COMPONENT_MANAGER = mockManager({
+      globalStorage: new GlobalStorage(),
+      persistentStorage: new PersistentStorage()
+    });
+
+    COMPONENT_MANAGER.getComponentNamesForComponentTypes = () => [];
+  });
+
+  it('tab links contain the referrerPageUrl from global storage', () => {
+    COMPONENT_MANAGER.core.globalStorage.set(StorageKeys.REFERRER_PAGE_URL, 'yext.com');
+
+    const component = COMPONENT_MANAGER.create('Navigation', defaultConfig);
+    const wrapper = mount(component);
+
+    const navItem = wrapper.find('.js-yxt-navItem').first();
+    const tabLink = navItem.prop('href');
+
+    expect(tabLink).toContain('referrerPageUrl=yext.com');
+  });
+
+  it('tab links contain the context from global storage', () => {
```

Comment:
Could we also make a test that handles `setContext` being called after the component is initially mounted? In that case, the links should be re-rendered to include the new context.

Reply:
Sure. I don't think the tests have access to ANSWERS.setContext(), however can try it with globalStorage.set(StorageKeys.API_CONTEXT).